### PR TITLE
Security Provider cleanup

### LINKED ISSUE #19741: Clean OpenSea Security Provider Logic
### Describe the bug

The OpenSea security provider is planned to be deprecated and replaced. Cleaning the code a little will help us implement the new security provider and potentially help us maintain both as we transition.

Affected Components:

Confirm Pages
Confirm Approve
Signature Requests
Signature Request Original
Signature Request SIWE
Token Allowance

### Steps to reproduce

Turn on security provider setting. Test on relevant transactions mentioned above under "Affected components" 

### Error messages or log output

_No response_

### Version

10.32

### Build type

None

### Browser

Chrome

### Operating system

MacOS

### Hardware wallet

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by digiwand**
## Explanation

no logic updates. just refactoring the OpenSea security provider logic. 

The OpenSea security provider is planned to be deprecated and replaced. Cleaning the code a little will help us implement the new security provider and potentially help us maintain both as we transition. 

- Created a util function to abstract conditional logic - `isSuspiciousSecurityProviderResponse`
- Moved `SECURITY_PROVIDER_MESSAGE_SEVERITIES` constant to a new `shared/constants` file
  - previously, the constant  lived in a component folder. The problem was it was used outside of the component.
- Renamed `SECURITY_PROVIDER_MESSAGE_SEVERITIES` -> `SECURITY_PROVIDER_MESSAGE_SEVERITY`
- The conditional with a little difficult for me to understand. I updated the code to hopefully be a bit more understandable.
- Add test for new util function

Affected Components: 
- Confirm Pages
- Confirm Approve
- Signature Requests
- Signature Request Original
- Signature Request SIWE 
- Token Allowance

--
Fixes: https://github.com/MetaMask/metamask-extension/issues/19741
Relates to: https://github.com/MetaMask/MetaMask-planning/issues/558
Relates to: #17662

## Pre-merge author checklist

- [ ] I've clearly explained:
  - [ ] What problem this PR is solving
  - [ ] How this problem was solved
  - [ ] How reviewers can test my changes
- [ ] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [ ] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [ ] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

### CODE REVIEW

**Review by blackdevelopa — APPROVED** [review]
Looks good to me

File: ui/components/app/confirm-page-container/confirm-page-container-content/confirm-page-container-content.component.js [hunk 1]
```diff
@@ -214,15 +215,11 @@ export default class ConfirmPageContainerContent extends Component {
...
-          txData?.securityProviderResponse?.flagAsDangerous !==
-            SECURITY_PROVIDER_MESSAGE_SEVERITIES.NOT_MALICIOUS) ||
-        (txData?.securityProviderResponse &&
-          Object.keys(txData.securityProviderResponse).length === 0) ? (
+        {isSuspiciousResponse(txData?.securityProviderResponse) && (
           <SecurityProviderBannerMessage
             securityProviderResponse={txData.securityProviderResponse}
           />
-        ) : null}
+        )}
```
Comment (jpuri) [inline ui/components/app/confirm-page-container/confirm-page-container-content/confirm-page-container-content.component.js:222]: Another approach could be moving this condition `isSuspiciousResponse(txData?.securityProviderResponse)` inside `SecurityProviderBannerMessage` component.

**Review by jpuri — APPROVED** [review]
Nice cleanup 👍

File: ui/components/app/confirm-page-container/confirm-page-container-content/confirm-page-container-content.component.js (hunk 1 above)
Comment (digiwand) [reply ui/components/app/confirm-page-container/confirm-page-container-content/confirm-page-container-content.component.js:222]: I like this suggestion. We can apply it to the new security provider code

**Pull request merged**